client/verta/verta/_tracking/experimentrun.py

```diff
@@ -226,32 +226,62 @@ def _log_artifact(self, key, artifact, artifact_type, extension=None, method=Non
+
+                # check if multipart upload was in progress
+                url = "{}://{}/api/v1/modeldb/experiment-run/getCommittedArtifactParts".format(
+                    self._conn.scheme,
+                    self._conn.socket,
+                )
+                msg = _CommonService.GetCommittedArtifactParts(id=self.id, key=key)
+                data = _utils.proto_to_json(msg)
+                response = _utils.make_request("GET", url, self._conn, params=data)
+                if response.status_code == 404:  # old backend without multipart support
+                    # TODO: check if this also happens when artifact was not multipart
+                    raise conflict_error
+                elif not response.ok:
+                    _utils.raise_for_http_error(response)
+
+                # get last part number
+                response_msg = _utils.json_to_proto(response.json(), _CommonService.GetCommittedArtifactParts.Response)
+                if not response_msg.artifact_parts:
+                    url_for_artifact = self._get_url_for_artifact(key, "PUT", part_num=1)
+
+                    if url_for_artifact.multipart_upload_ok:  # there should be parts when completed
+                        last_part_num = 0
+                    else:
+                        raise conflict_error
+                else:
+                    last_part_num = max(part.part_number for part in response_msg.artifact_parts)
+
+                # resume upload
+                self._upload_artifact(key, artifact_stream, start_part_num=last_part_num + 1)
             else:
                 _utils.raise_for_http_error(response)
-
-        if VERTA_ARTIFACT_DIR:
-            print("logging artifact")
-            with open(artifact_path, 'wb') as f:
-                shutil.copyfileobj(artifact_stream, f)
-            print("log complete; file written to {}".format(artifact_path))
         else:
-            self._upload_artifact(key, artifact_stream)
+            if VERTA_ARTIFACT_DIR:
+                print("logging artifact")
+                with open(artifact_path, 'wb') as f:
+                    shutil.copyfileobj(artifact_stream, f)
+                print("log complete; file written to {}".format(artifact_path))
+            else:
+                self._upload_artifact(key, artifact_stream)
 
         self._clear_cache()
 
-    def _upload_artifact(self, key, artifact_stream, part_size=64*(10**6)):
+    def _upload_artifact(self, key, artifact_stream, start_part_num=1):
         """
         Uploads `artifact_stream` to ModelDB artifact store.
 
         Parameters
         ----------
         key : str
         artifact_stream : file-like
-        part_size : int, default 64 MB
-            If using multipart upload, number of bytes to upload per part.
+        start_part_num : int, default 1
+            If using multipart upload, what part number to start uploading with.
 
         """
         # TODO: add to Client config
```

Comment:
🚨 See the next seven lines: we actually change `part_size` through an environment variable to accommodate a test ([this one](https://github.com/VertaAI/modeldb/blob/8831bd9/client/verta/tests/test_artifacts.py#L106-L113), makes `part_size` smaller so the test runs faster)

Could you implement a way that checks both `_artifact_utils.MULTIPART_UPLOAD_PART_SIZE` and `os.environ.get('VERTA_ARTIFACT_PART_SIZE', "")`? 🙇 

This may be a bigger problem because part size can change, but this env var is undocumented so it shouldn't be a problem for now (please also add a comment about this, if it makes sense to you!).

Reply:
Oh, I realized now that part_size wasn't used in this code at all. I've fixed it now. What do you mean by checks both of them? You mean that VERTA_ARTIFACT_PART_SIZE env wasn't really used here or something else?

I added a comment that it is used in test

Reply:
>What do you mean by checks both of them?

I mean: please implement something like
```python
if os.environ.get('VERTA_ARTIFACT_PART_SIZE', ""):
    part_size = 
else:
    part_size = _artifact_utils.MULTIPART_UPLOAD_PART_SIZE
```
but you've already done that! 👍